src/fileimpl.cpp

```diff
@@ -105,21 +105,40 @@ sectionSubReader(const FileReader& zimReader, const std::string& sectionName,
         new DirectDirentAccessor(direntReader, std::move(urlPtrReader), entry_index_t(header.getArticleCount())));
 
 
-    auto titleIndexReader = sectionSubReader(*zimReader,
-                                        "Title index table",
-                                        offset_t(header.getTitleIdxPos()),
-                                        zsize_t(4*header.getArticleCount()));
-
-    mp_titleDirentAccessor.reset(
-        new IndirectDirentAccessor(mp_urlDirentAccessor, std::move(titleIndexReader), title_index_t(header.getArticleCount())));
-
     clusterOffsetReader = sectionSubReader(*zimReader,
                                            "Cluster pointer table",
                                            offset_t(header.getClusterPtrPos()),
                                            zsize_t(8*header.getClusterCount()));
 
     quickCheckForCorruptFile();
 
+    offset_t titleOffset(header.getTitleIdxPos());
+    zsize_t  titleSize(4*header.getArticleCount());
+
+    for (auto url:{"listing/titleOrdered/v1", "listing/titleOrdered/v0"}) {
+      auto result = direntLookup().find('X', url);
+      if (result.first) {
+        auto dirent = mp_urlDirentAccessor->getDirent(result.second);
+        auto cluster = getCluster(dirent->getClusterNumber());
+        if (cluster->isCompressed()) {
+          // This is a ZimFileFormatError.
+          // Let's be tolerent and skip the entry
+          continue;
+        }
+        titleOffset = getClusterOffset(dirent->getClusterNumber()) + offset_t(1) + cluster->getBlobOffset(dirent->getBlobNumber());
```

Comment:
It would be nice of `zim::Cluster::getBlobOffset()` was in charge of the 1 (the cluster type byte) required in this computation of the full offset relative to the beginning of the ZIM archive. That would make it more consistent with `zim::Cluster::getOffset()`. Only a single pre-existing usage of `zim::Cluster::getBlobOffset()` in `FileImpl::getBlobOffset()` needs to be updated, in addition to this new code.

Reply:
Yes. Historically the first byte was parsed by the fileimpl/reader and the cluster was created on the "real" data reader. So the offset was off one.
But now, the cluster is directly initialized from it offset. It is better that the cluster handles it itself.

This is done in a new commit (at beginning of the branch).